docs/data-sources/scope.md

```diff
@@ -0,0 +1,28 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "boundary_scope Data Source - terraform-provider-boundary"
+subcategory: ""
+description: |-
+  The scope data source allows you to discover an existing Boundary scope by name.
+---
+
+# boundary_scope (Data Source)
+
+The scope data source allows you to discover an existing Boundary scope by name.
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The name of the scope to retrieve.
+- `scope_id` (String) The parent scope ID that will be queried for the scope.
```

Comment:
Can we make this more explicitly the parent scope? How about
```suggestion
- `parent_scope_id` (String) The parent scope ID that will be queried for the scope.
```
This is consistent with how we represent this over the API: https://github.com/hashicorp/boundary/blob/19180af0ebdf5f22040b045d010aa8cedb858a80/internal/proto/controller/api/resources/scopes/v1/scope.proto#L26

Reply:
This sounds like a sensible change. I looked at the [scope resource](https://registry.terraform.io/providers/hashicorp/boundary/latest/docs/resources/scope) example to guide me on this, but it likely makes sense to implement this here and then make a change to the resource for consistency across the provider to match the API.

Reply:
This is not quite correct. `ScopeInfo` is extra information about the scope referenced in the `scope_id` field of a resource output. `parent_scope_id` is _never_ used on input (nor is `ScopeInfo`, which is the only place that contains `parent_scope_id`).